src/gwt/panmirror/src/editor/src/api/widgets/text.tsx

```diff
@@ -0,0 +1,55 @@
+/*
+ * text.tsx
+ *
+ * Copyright (C) 2019-20 by RStudio, PBC
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+import React, { ChangeEventHandler, KeyboardEventHandler } from 'react';
+
+import { WidgetProps } from './react';
+
+import './text.css';
+
+export interface TextInputProps extends WidgetProps {
+  widthChars: number;
+  tabIndex?: number;
+  className?: string;
+  placeholder?: string;
+  iconAdornment? : string;
+  onChange?: ChangeEventHandler;
+  onKeyDown?: KeyboardEventHandler<HTMLInputElement>;
+  onKeyUp?: KeyboardEventHandler<HTMLInputElement>;
+}
+
+export const TextInput = React.forwardRef<any, TextInputProps>((props, ref) => {
```

Comment:
I'm surprised to see `any` in the type signature here, does `HTMLInputElement` not work?

Reply:
Nope just an error on my part (similar error in select.tsx as well). Resolved thx for catching!